### PR TITLE
fix: CI/CD

### PR DESCRIPTION
**Pull request by scottarc**
*Issue #, if available:*

*Description of changes:*

This aims to fix the unit tests in the CI/CD pipeline to unblock other changes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -12,7 +12,7 @@ envlist =
 # linters :: Runs all linters over all source code.
 # linters-tests :: Runs all linters over all tests.
 
-ignore_basepython_conflict = True
+# ignore_basepython_conflict = True
```
Comment (farleyb-amazon) [inline tox.ini:15]: Minor: can we either remove this, or add a comment explaining why I might want to sometimes comment it out?

File: tox.ini [hunk 2]
```diff
@@ -323,3 +323,4 @@ commands =
     # Unsetting the TWINE_REPOSITORY_URL defaults twine to using production PyPI
     unset TWINE_REPOSITORY_URL
```
Comment (farleyb-amazon) [inline tox.ini:324]: If i understand the below change correctly, by setting `--repository pypi` we're explicitly telling it to go to prod pypi. In which case we can remove the (broken) `unset` line here.

File: tox.ini [hunk 3]
```diff
@@ -311,6 +310,7 @@ passenv =
 setenv =
     TWINE_REPOSITORY_URL = https://test.pypi.org/legacy/
```
Comment (farleyb-amazon) [inline tox.ini:311]: Actually should we remove this as well? Just to remove confusion about how we're specifying the repository.

File: tox.ini (hunk 1 above)
Comment (scottarc) [reply tox.ini:15]: I think we can safely remove it.

File: tox.ini (hunk 2 above)
Comment (farleyb-amazon) [reply tox.ini:322]: Update: I was wrong -- `TWINE_REPOSITORY_URL` overrides `--repository`, so we still want to unset it.

Comment (scottarc) [reply tox.ini:322]: https://github.com/aws/aws-encryption-sdk-cli/pull/185/commits/82a3dc65ece1768e05380d0bf0d9e02f74725ee2 :)

Being wrong is often the first step to learning something new :D

File: tox.ini [hunk 4]
```diff
@@ -294,23 +292,41 @@ deps =
...
+    TWINE_REPOSITORY_URL
+setenv =
+    # Explicitly set the URL as the env variable value, which will cause us to
+    # throw an error if the variable is not set. Otherwise, omission of the
+    # env variable could cause us to unintentionally upload to the wrong repo
+    TWINE_REPOSITORY_URL = {env:TWINE_REPOSITORY_URL}
+commands =
+    {[testenv:release-base]commands}
+    # Omitting an explicit repository will cause twine to use the repository
+    # specified in the environment variable
```
Comment (farleyb-amazon) [inline tox.ini:319]: Need the `twine` command here as well since we're not setting it in the base target anymore.

File: tox.ini [hunk 5]
```diff
@@ -294,23 +292,41 @@ deps =
...
+    # as this overrides other ways of setting the repository and could
+    # unexpectedly result in releasing to the wrong repo
     {[testenv]passenv} \
     TWINE_USERNAME \
-    TWINE_PASSWORD \
-    TWINE_REPOSITORY_URL
+    TWINE_PASSWORD
 commands =
     {[testenv:build]commands}
-    twine upload --skip-existing --repository testpypi dist/*
```
Comment (mattsb42-aws) [inline tox.ini:303]: I would argue that the correct fix here is to remove `--repository testpypi`. That was a miss on my part when I initially populated out these changes.

The `--repository` flag identifies the profile in `~/.pypirc` that should be used. If you are specifying your username and repo url through environment variables, that flag is unnecessary.

Historical note: This and other Crypto Tools Python repos used be configured to use the `~/.pypirc` file for `twine` configuration. More recently I made changes to set the appropriate values via environment variables instead, both to remove unclear dependency on configuration that exists outside of the repo and to allow for future automation (ex passing a token in via GHS or `secrets-helper`).

**Review by farleyb-amazon — APPROVED** [review]

**Pull request merged**